code/datums/components/quality.dm

```diff
@@ -0,0 +1,123 @@
+#define UNUSABLE_QUALITY 0.7
+#define SHODDY_QUALITY 0.8
+#define POOR_QUALITY 0.9
+#define NORMAL_QUALITY 1
+#define GOOD_QUALITY 1.05
+#define FINE_QUALITY 1.1
+#define SUPERIOR_QUALITY 1.2
+#define EXCEPTIONAL_QUALITY 1.3
+#define ARTISAN_QUALITY 1.35
+#define MASTERWORK_QUALITY 1.4
+
+#define ARTIFACT_QUALITY 1.4
+
+
+///This Component handles adding quality to items
+/datum/component/quality
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/list/quality_levels = list(UNUSABLE_QUALITY,SHODDY_QUALITY,POOR_QUALITY,NORMAL_QUALITY,GOOD_QUALITY,FINE_QUALITY,SUPERIOR_QUALITY,EXCEPTIONAL_QUALITY,ARTISAN_QUALITY,MASTERWORK_QUALITY,ARTIFACT_QUALITY )
+	///Normal distribution
+	var/quality_level
+
+	var/mob/living/carbon/human/creator
+	var/datum/mind/creator_mind
+
+	var/obj/item/old_parent_item
+
+
+///_Creator - mob that is the owner of the item, _skill - skill that the quality should be based off of,_quality_list - custom distribution optional, defaults to normal distribution
+/datum/component/quality/Initialize(mob/living/carbon/human/_creator,datum/skill/_skill,_quality_val)
+	if(!isitem(parent) || !_creator)
+		return COMPONENT_INCOMPATIBLE
+	var/quality_val
+
+	if(_quality_val != null)
+		quality_val = _quality_val
+	else
+		quality_val = pick(2;0 ,4;1 ,6;2 ,4;3 ,2;4) // mormal distribution
+	creator = _creator
+
+	var/quality_bracket = creator.mind.get_skill_modifier(_skill, SKILL_QUALITY_MODIFIER)
+	creator_mind =  creator?.mind
+	var/obj/item/parent_item = parent
+
+	parent_item.has_quality = TRUE
+
+	old_parent_item = parent_item
```

Comment:
I don't really see the purpose of this yet, they are the same var throughout the code?